src/main/java/org/icgc/argo/program_service/services/ego/EgoService.java

```diff
@@ -245,13 +246,14 @@ public void initAdmin(String adminEmail, String shortName) {
 
   public List<User> getUsersInGroup(String programShortName) {
     val userResults = new ArrayList<User>();
-
     programEgoGroupRepository.findAllByProgramShortName(programShortName).forEach(programEgoGroup -> {
       val groupId = programEgoGroup.getEgoGroupId();
+      val role = programEgoGroup.getRole();
       try {
         val egoUserStream = egoClient.getUsersByGroupId(groupId);
-        egoUserStream.map(programConverter::egoUserToUser)
-          .forEach(userResults::add);
+        egoUserStream
+                .map(egoUser -> egoUser.setRole(role))
+                .map(programConverter::egoUserToUser).forEach(userResults::add);
```

Comment:
when this method is called, it is unclear if the request group has 0 users or if the group does not exist or if there is a database error in ego, because in all cases an empty list will be returned which can be very misleading and hard to debug. 


My suggestion is to log the exception (as is already being done), but also throw it and handle it else where. If that is done, the code can be refactored by replacing this block of code with

```java
return egoClient.getUsersByGroupId(groupId)
    .map(u -> u.setRole(role))
    .map(programConverter::egoUserToUser)
    .collect(Collectors.toUnmodifiableList());
```
since `userResults` is not being used anywhere else 
and delete the `val userResults = new ArrayList<User>();` line